glusterd2/xlator/load.go

```diff
@@ -83,17 +83,14 @@ func loadXlator(xlPath string) (*Xlator, error) {
 	defer C.dlclose(handle)
 
 	xl := new(Xlator)
-	xl.ID = strings.TrimSuffix(filepath.Base(xlPath), filepath.Ext(xlPath))
 
 	xlSym := C.CString("xlator_api")
 	defer C.free(unsafe.Pointer(xlSym))
 
 	p := C.dlsym(handle, xlSym)
 	if p != nil {
 		xp := (*C.xlator_api_t)(p)
-		// FIXME: It's named "server-protocol" instead of "server" in server.so
-		//        https://review.gluster.org/18879
-		// xl.ID = C.GoString(xp.identifier)
+		xl.ID = C.GoString(xp.identifier)
```

Comment:
question:
can we add the check here itself, in simplified state

```
if xl.ID = C.GoString(xp.identifier);xl.ID==""{
                xl.ID = strings.TrimSuffix(filepath.Base(xlPath),
		filepath.Ext(xlPath))
}
```

Reply:
The code below is common to both if and else part.